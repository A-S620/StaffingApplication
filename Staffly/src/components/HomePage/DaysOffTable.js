import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';
import {connect} from "react-redux";

const columns = [
    {id: 'TypeId', label: 'Type ID', minWidth: 50},
    {id: 'PersonId', label: 'Person ID', minWidth: 100},
    {id: 'RequestDate', label: 'Request Date', minWidth: 170},
    {id: 'StartDate', label: 'Start Date', minWidth: 170},
    {id: 'EndDate', label: 'End Date', minWidth: 170},
    {id: 'Comment', label: 'Comment', minWidth: 170},
];



const useStyles = makeStyles({
    root: {
        // width: '100%',
    },
    container: {
        maxHeight: 240,
        // maxWidth: 700,
    },
});

function DaysOffTable(props) {
    const classes = useStyles();
    const [page, setPage] = React.useState(0);
    const [rowsPerPage, setRowsPerPage] = React.useState(10);

    const handleChangePage = (event, newPage) => {
        setPage(newPage);
    };

    const handleChangeRowsPerPage = (event) => {
        setRowsPerPage(+event.target.value);
        setPage(0);
    };

    return (
        <Paper className={classes.root}>
            <TableContainer className={classes.container}>
                <Table stickyHeader aria-label="sticky table">
                    <TableHead>
                        <TableRow>
                            {columns.map((column) => (
                                <TableCell
                                    key={column.id}
                                    align={column.align}
                                    style={{minWidth: column.minWidth}}
                                >
                                    {column.label}
                                </TableCell>
                            ))}
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {props.daysOff.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((row) => {
                            return (
                                <TableRow hover role="checkbox" tabIndex={-1} key={row.RequestId}>
                                    {columns.map((column) => {
                                        const value = row[column.id];
                                        return (
                                            <TableCell key={column.id} align={column.align}>
                                                {column.format && typeof value === 'number' ? column.format(value) : value}
                                            </TableCell>
                                        );
                                    })}
                                </TableRow>
                            );
                        })}
                    </TableBody>
                </Table>
            </TableContainer>
            <TablePagination
                rowsPerPageOptions={[10, 25, 100]}
                component="div"
                count={props.daysOff.length}
                rowsPerPage={rowsPerPage}
                page={page}
                onChangePage={handleChangePage}
                onChangeRowsPerPage={handleChangeRowsPerPage}
            />
        </Paper>
    );
}

const mapStateToProps = state => {
    return {
        daysOff: state.daysoff
    }
}
export default connect(mapStateToProps, {})(DaysOffTable);