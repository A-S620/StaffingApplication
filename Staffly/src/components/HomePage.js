import React from 'react';
import ShiftsTable from "./HomePage/ShiftsTable";
import DaysOffTable from "./HomePage/DaysOffTable";
import {Grid, makeStyles, Typography, Button} from "@material-ui/core";
import {Container} from "@material-ui/core"

const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%'
    },
    container: {
        width: '40%'
    },
    alignItemsAndJustifyContent: {
        width: "100%",
        height: "100%",
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'pink',
    },

}));

export default function HomePage() {
    const classes = useStyles();
    return (
        <Grid className={classes.root}
                   direction="column"
                   justify="flex-start"
                   alignItems="center">
            <Container style={{marginBottom: 20}} >
                <Typography variant="h6" style={{marginBottom: 5}}>
                    Days Off:
                </Typography>


                <DaysOffTable/>

            </Container>
            <Container>
                <Typography variant="h6" style={{marginBottom: 5}}>
                    Shifts:
                </Typography>
                <ShiftsTable/>

            </Container>

        </Grid>
    );
}

