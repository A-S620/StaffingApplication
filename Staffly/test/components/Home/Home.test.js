import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux'
import HomePage from "../../../src/components/HomePage";
describe('Home', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<HomePage debug />);

        expect(component).toMatchSnapshot();
    });
});