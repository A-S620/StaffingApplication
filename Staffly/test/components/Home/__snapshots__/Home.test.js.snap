// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly in "debug" mode 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-root-1"
  direction="column"
  justify="flex-start"
>
  <WithStyles(ForwardRef(Container))
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "marginBottom": 5,
        }
      }
      variant="h6"
    >
      Days Off:
    </WithStyles(ForwardRef(Typography))>
    <Connect(DaysOffTable) />
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "marginBottom": 5,
        }
      }
      variant="h6"
    >
      Shifts:
    </WithStyles(ForwardRef(Typography))>
    <Connect(ShiftsTable) />
  </WithStyles(ForwardRef(Container))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-root-1"
  direction="column"
  justify="flex-start"
>
  <WithStyles(ForwardRef(Container))
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "marginBottom": 5,
        }
      }
      variant="h6"
    >
      Days Off:
    </WithStyles(ForwardRef(Typography))>
    <Connect(DaysOffTable) />
  </WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "marginBottom": 5,
        }
      }
      variant="h6"
    >
      Shifts:
    </WithStyles(ForwardRef(Typography))>
    <Connect(ShiftsTable) />
  </WithStyles(ForwardRef(Container))>
</WithStyles(ForwardRef(Grid))>
`;
